featuretools/tests/synthesis/test_deep_feature_synthesis.py

```diff
@@ -1195,3 +1208,84 @@ def test_primitive_options_multiple_inputs(es):
             assert 'datetime' in variables
             if len(variables) == 2:
                 assert 'value' != variables[0]
+
+    options2 = {Trend: [{'include_entities': ['log'],
+                         'ignore_variables': {'log': ['value']}},
+                        {'include_entities': ['log'],
+                         'include_variables': {'log': ['datetime']}}]}
+    dfs_obj2 = DeepFeatureSynthesis(target_entity_id='sessions',
+                                    entityset=es,
+                                    agg_primitives=['trend'],
+                                    trans_primitives=[],
+                                    primitive_options=options2)
+    features2 = dfs_obj2.build_features()
+
+    assert set(features2) == set(features1)
+
+
+def test_primitive_options_class_names(es):
+    options1 = {
+        'mean': {'include_entities': ['customers']}
+    }
+
+    options2 = {
+        Mean: {'include_entities': ['customers']}
+    }
+
+    dfs_obj1 = DeepFeatureSynthesis(target_entity_id='cohorts',
+                                    entityset=es,
+                                    agg_primitives=['mean'],
+                                    trans_primitives=[],
+                                    primitive_options=options1)
+    features1 = dfs_obj1.build_features()
+
+    dfs_obj2 = DeepFeatureSynthesis(target_entity_id='cohorts',
+                                    entityset=es,
+                                    agg_primitives=['mean'],
+                                    trans_primitives=[],
+                                    primitive_options=options2)
+    features2 = dfs_obj2.build_features()
+
+    dfs_obj3 = DeepFeatureSynthesis(target_entity_id='cohorts',
+                                    entityset=es,
+                                    agg_primitives=[Mean],
+                                    trans_primitives=[],
+                                    primitive_options=options1)
+    features3 = dfs_obj3.build_features()
+
+    dfs_obj4 = DeepFeatureSynthesis(target_entity_id='cohorts',
+                                    entityset=es,
+                                    agg_primitives=[Mean],
+                                    trans_primitives=[],
+                                    primitive_options=options2)
+    features4 = dfs_obj4.build_features()
```

Comment:
Maybe we simplify this a little by looping over the four options. Something like this:
```
primitives = [Mean, 'mean']
options = [options1, options2]
results = []
for primitive in primitives:
    for option in options:
        features.append(DeepFeatureSynthesis(...).build_features())

```
And then update the assertions that follow accordingly to use the items in the results list as needed.